File: tools/lua-bindings/autobuild/jiguang.lua
-- AUTO BUILD, DON'T MODIFY!

dofile "autobuild/jiguang-types.lua"

name = "jiguang"
path = "../../frameworks/plugins/jiguang"
headers = [[
    #include "lua-bindings/lua_conv.h"
    #include "lua-bindings/lua_conv_manual.h"
    #include "jiguang.h"
]]
chunk = nil
luaopen = nil


typeconf 'cclua::plugin::jpush'
    .supercls(nil)
    .reg_luatype(true)
    .chunk(nil)
    .luaopen([[cclua::runtime::registerFeature("cclua.plugin.jpush.ios", true);]])
    .indexerror(nil)
    .ifdef('*', '#ifdef CCLUA_BUILD_JPUSH')
    .func(nil, 'static void init(const std::string &appKey, const std::string &channel)')
    .func(nil, 'static void setAlias(const std::string &alias)')
    .func(nil, 'static void deleteAlias()')
    .func(nil, 'static void addTags(const std::set<std::string> &tags)')
    .func(nil, 'static void setTags(const std::set<std::string> &tags)')
    .func(nil, 'static void deleteTags(const std::set<std::string> &tags)')
    .func(nil, 'static void cleanTags()')
    .func(nil, 'static void setDebug(bool enabled)')
    .func(nil, 'static void setLogOFF()')
    .func(nil, 'static void setBadge(int value)')
    .func(nil, 'static bool isEnabled()')
    .func(nil, 'static void requestPermission()')
    .func(nil, 'static std::string getRegistrationID()')
    .prop('enabled', nil, nil)
    .prop('registrationID', nil, nil)

typeconf 'cclua::plugin::jauth'
    .supercls(nil)
    .reg_luatype(true)
    .chunk(nil)
    .luaopen([[cclua::runtime::registerFeature("cclua.plugin.jauth.ios", true);]])
    .indexerror(nil)
    .ifdef('*', '#ifdef CCLUA_BUILD_JAUTH')
    .func(nil, 'static void init(const std::string &appKey, const std::string &channel)')
    .func(nil, 'static bool isInitSuccess()')
    .func(nil, 'static void setDebug(bool enabled)')
    .func(nil, 'static bool checkVerifyEnable()')
    .func(nil, 'static void clearPreLoginCache()')
    .func(nil, 'static void dismissLoginAuth(@optional bool needCloseAnim)')
    .func(nil, 'static void setSmsIntervalTime(int64_t intervalTime)')
    .func(nil, 'static void configUI(cocos2d::ValueMap &value, @optional bool landscape)')
    .func(nil, 'static void dispatch(const std::string &event, const cocos2d::ValueMap &data)')
    .callback {
        funcs =  {
            'static void getToken(int timeout)'
        },
        tag_maker = 'Token',
        tag_mode = 'new',
        tag_store = 0,
        tag_scope = 'once',
    }
    .callback {
        funcs =  {
            'static void preLogin(int timeout)'
        },
        tag_maker = 'preLogin',
        tag_mode = 'new',
        tag_store = 0,
        tag_scope = 'once',
    }
    .callback {
        funcs =  {
            'static void loginAuth(int timeout)'
        },
        tag_maker = 'loginAuth',
        tag_mode = 'new',
        tag_store = 0,
        tag_scope = 'once',
    }
    .callback {
        funcs =  {
            'static void getSmsCode(const std::string &phonenum, const std::string &signid, const std::string &tempid)'
        },
        tag_maker = 'SmsCode',
        tag_mode = 'new',
        tag_store = 0,
        tag_scope = 'once',
    }
    .callback {
        funcs =  {
            'static void setDispatcher(@localvar const cclua::Callback &dispatcher)'
        },
        tag_maker = 'Dispatcher',
        tag_mode = 'replace',
        tag_store = 0,
        tag_scope = 'object',
    }
    .prop('initSuccess', nil, nil)

typeconf 'cclua::plugin::janalytics::EventType'
    .supercls(nil)
    .reg_luatype(true)
    .chunk(nil)
    .luaopen(nil)
    .indexerror('rw')
    .ifdef('*', '#ifdef CCLUA_BUILD_JANALYTICS')
    .enum('LOGIN', 'cclua::plugin::janalytics::EventType::LOGIN')
    .enum('REGISTER', 'cclua::plugin::janalytics::EventType::REGISTER')
    .enum('PURCHASE', 'cclua::plugin::janalytics::EventType::PURCHASE')
    .enum('BROWSE', 'cclua::plugin::janalytics::EventType::BROWSE')
    .enum('COUNT', 'cclua::plugin::janalytics::EventType::COUNT')
    .enum('CALCULATE', 'cclua::plugin::janalytics::EventType::CALCULATE')

typeconf 'cclua::plugin::janalytics'
    .supercls(nil)
    .reg_luatype(true)
    .chunk(nil)
    .luaopen([[cclua::runtime::registerFeature("cclua.plugin.janalytics.ios", true);]])
    .indexerror(nil)
    .ifdef('*', '#ifdef CCLUA_BUILD_JANALYTICS')
    .func(nil, 'static void init(const std::string &appKey, const std::string &channel)')
    .func(nil, 'static void startTrackPage(const std::string &pageName)')
    .func(nil, 'static void stopTrackPage(const std::string &pageName)')
    .func(nil, 'static void trackEvent(cclua::plugin::janalytics::EventType type, cocos2d::ValueMap &value)')
    .func(nil, 'static void identifyAccount(cocos2d::ValueMap &value)')
    .func(nil, 'static void detachAccount()')
    .func(nil, 'static void setFrequency(int frequency)')
    .func(nil, 'static void setDebug(bool enable)')

File: frameworks/plugins/jiguang/JiGuang.cpp
#include "jiguang.h"

#if defined(CCLUA_BUILD_JPUSH) && defined(CCLUA_OS_ANDROID)

USING_NS_CCLUA;
USING_NS_CCLUA_PLUGIN;

const char *jpush::JAVA_CLASS = "cclua/plugin/jiguang/JPush";

void jpush::init(const std::string &appKey, const std::string &channel)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "init", appKey, channel);
}

void jpush::setAlias(const std::string &alias)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setAlias", alias);
}

void jpush::deleteAlias()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "deleteAlias");
}

void jpush::addTags(const std::set<std::string> &tags)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "addTags", toJSONString(tags));
}

void jpush::setTags(const std::set<std::string> &tags)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setTags", toJSONString(tags));
}

void jpush::deleteTags(const std::set<std::string> &tags)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "deleteTags", toJSONString(tags));
}

void jpush::cleanTags()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "cleanTags");
}

void jpush::setDebug(bool enabled)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setDebugMode", enabled);
}

void jpush::setLogOFF()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setLogOFF");
}

void jpush::setBadge(int value)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setBadge", value);
}

bool jpush::isEnabled()
{
    return Jni::callStaticBooleanMethod(JAVA_CLASS, "isEnabled");
}

void jpush::requestPermission()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "requestPermission");
}

std::string jpush::getRegistrationID()
{
    return Jni::callStaticStringMethod(JAVA_CLASS, "getRegistrationID");
}

#endif // CCLUA_BUILD_JPUSH

#if defined(CCLUA_BUILD_JAUTH) && defined(CCLUA_OS_ANDROID)

USING_NS_CCLUA;
USING_NS_CCLUA_PLUGIN;

const char *jauth::JAVA_CLASS = "cclua/plugin/jiguang/JAuth";
Callback jauth::_dispatcher;

void jauth::init(const std::string &appKey, const std::string &channel)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "init", appKey, channel);
}

bool jauth::isInitSuccess()
{
    return Jni::callStaticBooleanMethod(JAVA_CLASS, "isInitSuccess");
}

void jauth::setDebug(bool enabled)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setDebug", enabled);
}

bool jauth::checkVerifyEnable()
{
    return Jni::callStaticBooleanMethod(JAVA_CLASS, "checkVerifyEnable");
}

void jauth::getToken(int timeout)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "getToken", timeout);
}

void jauth::preLogin(int timeout)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "preLogin", timeout);
}

void jauth::clearPreLoginCache()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "clearPreLoginCache");
}

void jauth::loginAuth(int timeout)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "loginAuth", timeout);
}

void jauth::dismissLoginAuth(bool needCloseAnim)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "dismissLoginAuth", needCloseAnim);
}

void jauth::getSmsCode(const std::string &phonenum, const std::string &signid, const std::string &tempid)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "getSmsCode", phonenum, signid, tempid);
}

void jauth::setSmsIntervalTime(int64_t intervalTime)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setSmsIntervalTime", intervalTime);
}

void jauth::configUI(cocos2d::ValueMap &value, bool landscape)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "configUI", toJSONString(value), landscape);
}
#endif // CCLUA_BUILD_JAUTH

#if defined(CCLUA_BUILD_JANALYTICS) && defined(CCLUA_OS_ANDROID)

USING_NS_CCLUA;
USING_NS_CCLUA_PLUGIN;

const char *janalytics::JAVA_CLASS = "cclua/plugin/jiguang/JAnalytics";

void janalytics::init(const std::string &appKey, const std::string &channel)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "init", appKey, channel);
}

void janalytics::startTrackPage(const std::string &pageName)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "startTrackPage", pageName);
}

void janalytics::stopTrackPage(const std::string &pageName)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "stopTrackPage", pageName);
}

void janalytics::trackEvent(EventType type, cocos2d::ValueMap &value)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "trackEvent", (int)type, toJSONString(value));
}

void janalytics::identifyAccount(cocos2d::ValueMap &value)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "identifyAccount", toJSONString(value));
}

void janalytics::detachAccount()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "detachAccount");
}

void janalytics::setFrequency(int frequency)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setFrequency", frequency);
}

void janalytics::setDebug(bool enable)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "setDebug", enable);
}
#endif // CCLUA_BUILD_JANALYTICS
